blazor/datagrid/how-to/customize-filter-popup-using-css.md

```diff
@@ -0,0 +1,96 @@
+---
+layout: post
+title: Customize filter popup using CSS in Blazor DataGrid | Syncfusion
+description: Learn here all about Customizing filter popup using CSS in Syncfusion Blazor DataGrid component and more.
+platform: Blazor
+control: DataGrid
+documentation: ug
+---
+
+# Customize Filter Popup Using CSS
+
+You can set the height and width of each column in the filter popup dialog using the CSS style.
+
+In the following sample, we have enabled an Excel-like filter by defining Type as `Excel`. You can set the height and width of each column in the pop-up filter based on the condition in the [OnActionBegin](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Grids.GridEvents-1.html#Syncfusion_Blazor_Grids_GridEvents_1_OnActionBegin) event of the Grid.
+
+```cshtml
+@using Syncfusion.Blazor.Grids
+
+<SfGrid ID= "Grid" DataSource="@Orders" AllowFiltering="true">
+    <GridEvents OnActionBegin="OnActionBegin" TValue="Order"></GridEvents>
+    <GridFilterSettings Type ="Syncfusion.Blazor.Grids.FilterType.Excel"></GridFilterSettings>
+    <GridColumns>
+        <GridColumn Field=@nameof(Order.OrderID) HeaderText="Order ID" TextAlign="TextAlign.Right" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(Order.CustomerID) HeaderText="Customer Name" Width="150"></GridColumn>
+        <GridColumn Field=@nameof(Order.OrderDate) HeaderText=" Order Date" Format="d" Type="ColumnType.Date" TextAlign="TextAlign.Right" Width="130"></GridColumn>
+        <GridColumn Field=@nameof(Order.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Right" Width="120"></GridColumn>
+    </GridColumns>
+</SfGrid>
+
+@if(IsLarge)
+{
+    <style>
+        #Grid .e-excelfilter.e-popup.e-popup-open {
+            height: 400px;
+            width: 350px !important;
+        }
+    </style>
+}
+@if(IsSmall)
+{
+    <style>
+        #Grid .e-excelfilter.e-popup.e-popup-open {
+            height: 450px;
+            width: 280px !important;
+        }
+    </style>   
+}
+
+@code{
+    public List<Order> Orders { get; set; }
+    public bool IsLarge { get; set; } = false;
+    public bool IsSmall { get; set; } = false;
+    public void OnActionBegin(ActionEventArgs<Order> Args)
+    {
+        if (Args.RequestType == Syncfusion.Blazor.Grids.Action.FilterBeforeOpen)
+        {
+            if(Args.ColumnName == "CustomerID")
+            {
+                IsLarge = true;
```

Comment:
how this isLArge and isSmall css is added to grid

Reply:
Based on this boolean value, we have applied css style here.
![image](https://user-images.githubusercontent.com/93311293/171619743-680b90d5-3b4d-453a-bbb6-bb680ff216cb.png)